### ISSUE TITLE
Split up Maven plugin into testable parts

### ISSUE THREAD

**Issue by helpermethod** [helpermethod/dependency-update-maven-plugin#12]
Because testing the plugin as a whole is a pretty hefty task, it's probably best to start splitting the plugin into multiple, testable parts.

Each part/class/module should receive their direct dependencies via constructor, so that dependencies can be mocked.

**Comment by helpermethod** [comment]
Some ideas:

* the `JGitProvider` is almost testable except for the authenticating part - maybe that one could be split out into its own class
* everything related to `updates` looks like it would be a candidate for its own class
* in `UpdateMojo.execute()` we could wire up all dependencies by either `manual DI` or some something like [koin](https://github.com/InsertKoinIO/koin)

**Comment by sparsick** [comment]
first both ideas match my ideas. I disagree with the third idea (DI) because Maven already uses a DI Framework called Plexus. I would prefer to stay on this path and use Plexus as DI framework and not to try introducing a new DI Framework.

**Comment by helpermethod** [comment]
Plexus is no longer the preferred approach for DI in Maven plugins, Sisu is https://wiki.eclipse.org/Sisu/PlexusMigration

**Comment by sparsick** [comment]
Yes you are right .. sorry I mismatch the names. My main point is to use the DI framework that Maven uses :)

**Comment by sparsick** [comment]
@georgberky and I try to write an own class for the update stuff today. We will send a proposal :)

**Comment by helpermethod** [comment]
Ah I also did! But send yours anyway ;).

Sandra Parsick <notifications@github.com> schrieb am Do., 2. Apr. 2020,
16:30:

**Comment by helpermethod** [comment]
Here you can find a first draft

https://github.com/helpermethod/dependency-update-maven-plugin/blob/issue-12/src/main/kotlin/com/github/helpermethod/UpdateResolver.kt

**Comment by helpermethod** [comment]
Hi @sparsick, @georgberky your code looks pretty good from a quick glance! If you want, you can continue extracting the update mechanic and I'll continue with something else.

**Comment by sparsick** [comment]
That would be a pleasure. As next step, we want to introduce unit tests for the single VersionUpdate classes. Maybe we will find a better name for them.
